tools/src/wyvern/tools/typedAST/core/expressions/Application.java

```diff
@@ -141,22 +160,31 @@ public FileLocation getLocation() {
 	public Expression generateIL(GenContext ctx, ValueType expectedType) {
 		CallableExprGenerator exprGen = function.getCallableExpr(ctx);
 		List<FormalArg> formals = exprGen.getExpectedArgTypes(ctx);
-		
+
+        int offset = 0;
 		// generate arguments		
 		List<Expression> args = new LinkedList<Expression>();
+        if(!this.generics.isEmpty()) {
+            for(String generic : this.generics) {
+                 args.add(new wyvern.target.corewyvernIL.expression.New(new TypeDeclaration(DefDeclaration.GENERIC_MEMBER, new NominalType("", generic), this.location)));
```

Comment:
> Instead of new NominalType("", generic), you should follow the code in UnresolvedType.getILType() to generate the ValueType properly.

Unfortunately this isn't working for me. The reason is the context isn't extended with the function arguments at the time the Value Type is resolved (i.e. when `getContainerForTypeAbbrev()` is called). It seems like a cyclical dependency: I need to call `getContainerForTypeAbbrev` to get the value type so I can add the argument to the argList, but I can't resolve `getContainerForTypeAbbrev` without having the value type already in the context.